ipatests/test_integration/test_cert.py

```diff
@@ -385,40 +392,44 @@ def test_rekey_request_id(self, request_cert):
                                           '-I', self.request_id])
         assert self.request_id in result.stdout_text
 
-    def test_rekey_keytype_DSA(self):
+    def test_rekey_keytype_DSA(self, request_cert):
         """Test certmonger rekey command works fine
 
         Test is to check if -G (keytype) with DSA fails
 
         related: https://bugzilla.redhat.com/show_bug.cgi?id=1249165
         """
-        result = self.master.run_command(
-            [
-                'ipa-getcert', 'request',
-                '-f', os.path.join(paths.OPENSSL_CERTS_DIR, "test_dsa.pem"),
-                '-k', os.path.join(paths.OPENSSL_PRIVATE_DIR, "test_dsa.key"),
-                '-K', 'test/{}'.format(self.master.hostname),
-            ]
-        )
-        req_id = re.findall(r'\d+', result.stdout_text)
-        status = tasks.wait_for_request(self.master, req_id[0], 100)
+        status = tasks.wait_for_request(self.master, self.request_id, 100)
         assert status == "MONITORING"
 
         # rekey with RSA key type
         self.master.run_command(['getcert', 'rekey',
-                                 '-i', req_id[0],
+                                 '-i', self.request_id,
                                  '-g', '3072',
                                  '-G', 'DSA'])
         time.sleep(100)
```

Comment:
You can replace the `sleep` with
```
status = tasks.wait_for_request(self.master, self.request_id, 100)
assert status == "MONITORING"
```
as it will be quicker.